spec/controllers/api/v1/notes_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require 'rails_helper'
+
+shared_examples 'successful response' do
+  it 'responds with the correct keys' do
+    expect(response_body.sample.keys).to eq(expected_keys)
+  end
+
+  it 'responds with 200 status' do
+    expect(response).to have_http_status(:ok)
+  end
+end
+
+describe Api::V1::NotesController, type: :controller do
+  describe 'GET #index' do
+    let(:user_note_count) { Faker::Number.between(from: 3, to: 10) }
+    let(:user_notes) { create_list(:note, user_note_count, user: user) }
+    let(:expected_keys) { %w[id title note_type content_length] }
+
+    context 'when user is logged in' do
+      include_context 'with authenticated user'
+
+      context 'when fetching all user notes' do
+        before do
+          user_notes
+          get :index
+        end
+
+        include_examples 'successful response'
+      end
+
+      context 'when fetching notes with page paramaters' do
+        let(:page) { 1 }
+        let(:page_size) { Faker::Number.between(from: 1, to: user_note_count) }
+
+        context 'when fetching notes with page size' do
+          before do
+            user_notes
+            get :index, params: { page_size: page_size }
+          end
+
+          include_examples 'successful response'
+
+          it 'retrive the correct number of notes' do
+            expect(response_body.size).to eq(page_size)
+          end
+        end
+
+        context 'when fetching notes with page and page size' do
+          before do
+            user_notes
+            get :index, params: { page: page, page_size: page_size }
+          end
+
+          include_examples 'successful response'
+        end
+      end
+
+      context 'when fetching notes using filters' do
+        let(:note_type) { %w[review critique].sample }
+        let(:filtered_notes) { user_notes.select { |note| note.note_type == note_type } }
+
+        context 'when note_type is valid' do
+          before do
+            user_notes
+            get :index, params: { note_type: note_type }
```

Comment:
Esto aplica para los tests en general, para no repetir tantas veces el get a lo largo de los context podemos parametrizar el valor de params, quedando algo asi:
```suggestion
            get :index, params: params
```
Redefiniendo params según el contexto lo necesite, y moviendo el `before { get :index, params: params }`  lo más arriba posible, cosa de escribirlo una sola vez :sunglasses:

Reply:
Cambie para usar before lo mas arriba posible y en cada contexto redifinir los parametros